### DISCUSSION TITLE
Plotting Datetime data

### DISCUSSION THREAD

**Discussion by pietz** [f5/unovis#220] [Q&A] [answered]
Hey team,

does unovis support plotting datetime data where the x-axis behaves similar to numeric values? I want to use bar charts, where the x-axis represents the time axis and the y-axis represents counts. I was wondering if unovis has this "intelligent" tick behavior not only for numeric values but datetime values as well.

Thanks,
Pietz

**Comment by rokotyan** [comment, marked as answer]
Yes, it's possible. By default, Unovis treats `Date` values or timestamps as regular numbers. I also recommend to use `Scale.scaleTime()` for your x-axis to make it produce nicer ticks, like in [this example](https://unovis.dev/gallery/view?collection=Lines%20and%20Areas&title=Multi%20Line%20Chart):

<img width="707" alt="image" src="https://github.com/f5/unovis/assets/755708/97da74e6-3581-42a3-a29c-88874ff61877">

**Reply by pietz** [reply]
brilliant, thanks!

**Reply by garbit** [reply]
How do you scale the domain to a given date range when you have no data? I have an empty array that defaults to 1970 if there's no data (see here: https://stackblitz.com/edit/dhx2xyzs?file=src%2FApp.vue)

I want to be able to show the chart between a given set of dates rather than from 1970.

**Reply by rokotyan** [reply]
@garbit You can use the `xDomain` config property of `XYContainer`
